src/components/computing-status/use-computing-status.ts

```diff
@@ -0,0 +1,151 @@
+import { UPDATE_TYPE_HEADER } from 'components/study-container';
+import { RunningStatus } from 'components/utils/running-status';
+import { UUID } from 'crypto';
+import { RefObject, useCallback, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { setComputingStatus } from 'redux/actions';
+import { ComputingType } from './computing-type';
+import { ReduxState, StudyUpdated } from 'redux/reducer.type';
+
+interface UseComputingStatusProps {
+    (
+        studyUuid: UUID,
+        nodeUuid: UUID,
+        fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+        invalidations: string[],
+        resultConversion: (x: string) => RunningStatus,
+        computingType: ComputingType
+    ): void;
+}
+
+interface LastUpdateProps {
+    studyUpdatedForce: StudyUpdated;
+    fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>;
+}
+
+function isWorthUpdate(
+    studyUpdatedForce: StudyUpdated,
+    fetcher: (studyUuid: UUID, nodeUuid: UUID) => Promise<string>,
+    lastUpdateRef: RefObject<LastUpdateProps>,
+    nodeUuidRef: RefObject<UUID>,
+    nodeUuid: UUID,
+    invalidations: Array<string>
```

Comment:
Maybe choose between `string[] `above or `Array<string> `here to be coherent ?

Reply:
fixed